### PR TITLE
Trivial: Remove 'LoginStateInterface' and replace with 'LoginState'

### PR DESCRIPTION
**Pull request by mtho11**
For consistency with the rest of the state reducers -- replacing 'LoginStateInterface' with 'LoginState'.

### CODE REVIEW

**Comment by jotak** [comment]
@mtho11 I changed it on purpose while fixing some issues reported by "lgtm" : https://lgtm.com/projects/g/kiali/kiali-ui/snapshot/dist-1506778376025-1547027795790/files/src/reducers/LoginState.ts?sort=name&dir=ASC&mode=heatmap#x22368d08b7a8fba2:1

Here LoginState exists here as a type and as a const. It's correct in term of ts language but maybe we shouldn't "hide" one with the other?
Note that the issue reported by lgtm is about assigning variable to a const, which isn't exactly what's going on here, so maybe it's a false positive.

**Comment by mtho11** [comment]
It looks like we don't have that issue on other reducers because the `const` version is lowercase:
```
import messageCenter from './MessageCenter';
import LoginState from './LoginState';
import HelpDropdownState from './HelpDropdownState';
import graphDataState from './GraphDataState';
import globalState from './GlobalState';
import namespaceState from './NamespaceState';
import UserSettingsState from './UserSettingsState';
import GrafanaState from './GrafanaState';
```
@jotak @aljesusg is acceptable that I rename the `const` from uppercase to lowercase to match the others and fix the issue that way?
So `LoginState` would be the `type` and `loginState` would be the `const`

**Comment by cfcosta** [comment]
IMHO: Type definitions are CamelCase and have higher priority than everything else. Instances with the same name than the type are a code smell, and should be renamed (or the type itself should be renamed if that's the case). Nothing starting with a lower case letter should be exported (kinda what Golang does by default). Adding hungarian notation to the variables only makes thing worse, because then we have something like `interface LoginStateInterface`. The compiler exists to help us make sense of those definitions, we should use it.

**Comment by mtho11** [comment]
@cfcosta I disagree with adopting `Go` practices in javascript/typescript. These languages are very different and should not mix styles.

> Adding hungarian notation to the variables only makes thing worse, because then we have something like interface LoginStateInterface. The compiler exists to help us make sense of those definitions, we should use it.
I do agree with this

**Comment by cfcosta** [comment]
> @cfcosta I disagree with adopting `Go` practices in javascript/typescript. These languages are very different and should not mix styles.

I don't mean to copy go, I mean that we do have to make a difference between local variables and exported items. If we're going to name them as methods/variables, we should at least wrap them and enforce `import X` instead of `import { x }` in those cases.

**Comment by jotak** [comment]
To me, having upper camel case for types/interfaces/classes and lower camel case for vars/consts, regardless they're exported or not, sounds more familiar (so basically, what @mtho11 suggests) but it's clear that we don't have currently a single style in our codebase. What is the standard in typical typescript projects?

**Comment by jotak** [comment]
> Adding hungarian notation to the variables only makes thing worse, because then we have something like interface LoginStateInterface. The compiler exists to help us make sense of those definitions, we should use it.

Just to be clear, the interface was not named "LoginStateInterface", it was still only "LoginState". "LoginStateInterface" was just a disambiguation import alias in the context of a file where LoginState was another thing.

**Comment by cfcosta** [comment]
> Just to be clear, the interface was not named "LoginStateInterface", it was still only "LoginState". "LoginStateInterface" was just a disambiguation import alias in the context of a file where LoginState was another thing.

This is true. Still disagree with the naming, but being a local alias makes the problem smaller ;)

**Comment by israel-hdez** [comment]
Here is a guide: https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md
Note that, apparently, there isn't a convention broadly accepted by the community. Most projects make their own code style, or adopt something. Some projects even adopt C# naming style.

**Comment by mtho11** [comment]
Here is the M$ Typescript style guide: https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names

**Comment by mtho11** [comment]
There is one use case for making functions start with upper case though -- when that function is representing a component or JSX snippet. Such as:

```
const Child = ({ match }) => (
  <div>
    <h3>ID: {match.params.id}</h3>
  </div>
)
```

This is the exception case, in my mind.

File: src/reducers/LoginState.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const INITIAL_LOGIN_STATE: LoginStateInterface = {
 };
 
 // This Reducer allows changes to the 'LoginState' portion of Redux Store
-const LoginState = (state: LoginStateInterface = INITIAL_LOGIN_STATE, action: KialiAppAction): LoginStateInterface => {
+const LoginState = (state: LoginState = INITIAL_LOGIN_STATE, action: KialiAppAction): LoginState => {
```
[the remark this answers is not in the document]
Comment (mtho11) [reply src/reducers/LoginState.ts]: @josejulio done

**Pull request merged**